bitmark-references/bitmark-node-software/bitmark-blockain.md

```diff
@@ -0,0 +1,166 @@
+
+The Transfer Record structure
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Type code         Byte       Transfer Record type code
+Link              Binary     Link to previous Issue or Transfer
+Owner             Account    The public key of the signer
+Signature         Signature  Ed25519 signature of linked previous owner
+CounterSignature  Signature  Ed25519 signature of this record owner
+
+
+## Transactions: Share
+
+
+## Transactions: Block ownership record
+
+
+## Blockchain structure
+
+### Block Header
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Version           U16        The version control block rules in action
+TransactionCount  U16        Number of transactions in the block
+Number            U64        The number of this block
+PreviousBlock     Argon2     Thirty two byte has of the previous block
+MerkleRoot        SHA3-256   Thirty two byte has of the transaction Merkle tree
+Timestamp         U64        Timestamp as a Unix (seconds after start of 1970-01-01)
+Difficulty        U64        Difficulty fraction as 57 bit mantissa + 8 bit exponent
+Nonce             U64        Nonce created by hashing to meet the difficulty
+
+### Merkle tree
+
+This is a way of combining hashes of a list of records to obtain a
+single hash such that only the same records in the same order will
+have the same hash.  It is done by making the hashes of all the
+records the final elements in a binary tree, with each pair hashed
+together to get the value above.  Any item not forming a pair at at
+some level of the tree if hashed with itself.  The final hash is known
+as the root.  In this system the SHA3-256 algorithm is used for the
+hashing process.
+
+The SHA3 algorithm is the current recommended hashing algorithm to use
+and fixes some vulnerability problems that were found in SHA2.  The
+SHA2 would have to be uses twice to protect against this and that
+costs more CPU resources.  The SHA3 is faster so reduces the time to
+build the Merkle Tree.  SHA1 has been broken so was not considered.
+
+
+### Block Hashing (argon2, difficulty, proof of work ...)
+
+To prevent hardware attack of the Proof of Work system using GPUs or
+ASICs a memory hard algorithm was used: Argon2.  Its parameters are
+set such that a GPU will be some orders of magnitude slower that a
+typical x86 CPU core.  At the settings uses the hashing power of CPUs
+is still very low on the order of a few hashes per second, this means
+that the difficulty scale is much lower than other blockchain systems.
+
+The difficulty is represented as a floating point value in the range:
+
+    1 ≥ d > 0
+
+The hash is considered as a 256 bit fixed point value with 8 bits
+before the decimal point:
+
+    [8 bits] . [248 bits]
+
+A hash meets the difficulty if it's value less than or equal to
+difficulty value.
+
+The difficulty value is encoded as a 57 bit mantissa, normalised so
```

Comment:
Could provide the link which explains floating-point format? https://en.wikipedia.org/wiki/Double-precision_floating-point_format